dbms/src/Flash/Planner/PlanType.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <common/types.h>
+
+namespace DB
+{
+struct PlanType
+{
+    enum PlanTypeEnum
+    {
+        Limit = 0,
+        TopN = 1,
+        Filter = 2,
+        Aggregation = 3,
```

Comment:
(just discussion) if someone wants to split `Aggregation` into `PartialAggregation` and `FinalAggregation`, do you think it's good to change here?

Reply:
My question is what's the right way to insert `LocalShuffle` into a plan?

Reply:
> (just discussion) if someone wants to split `Aggregation` into `PartialAggregation` and `FinalAggregation`, do you think it's good to change here?

LGTM

Reply:
> (just discussion) if someone wants to split `Aggregation` into `PartialAggregation` and `FinalAggregation`, do you think it's good to change here?

> (just discussion) if someone wants to split `Aggregation` into `PartialAggregation` and `FinalAggregation`, do you think it's good to change here?

Here is an attempt at splitting agg.
https://github.com/pingcap/tiflash/pull/4874

Reply:
> My question is what's the right way to insert `LocalShuffle` into a plan?

In `PhysicalPlan::build` or `optimize`?